File: DemoMMOUnity/Assets/Game/Net/Script/Net/CsProtocol/Protocol/Login/RegisterRequest.cs
using System.Collections.Generic;
using Newtonsoft.Json;
using ZJYFrameWork.Collection.Reference;
using ZJYFrameWork.Net.Core;
using ZJYFrameWork.Net.CsProtocol.Buffer;
using ZJYFrameWork.Spring.Utils;

namespace ZJYFrameWork.Net.CsProtocol
{
    public class RegisterRequest : Model, IPacket,IReference
    {
        public string account;
        public string password;
        public string affirmPassword;

        public static RegisterRequest ValueOf(string account, string affirmPassword, string password)
        {
            var packet = new RegisterRequest
            {
                account = account,
                affirmPassword = affirmPassword,
                password = password
            };
            return packet;
        }
        public static RegisterRequest ValueOf()
        {
            var packet = ReferenceCache.Acquire<RegisterRequest>();
            return packet;
        }

        public short ProtocolId()
        {
            return 1005;
        }

        public void Clear()
        {
            account = string.Empty;
            password = string.Empty;
            affirmPassword = string.Empty;
        }
    }


    public class RegisterRequestRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 1005;
        }

        public void Write(ByteBuffer buffer, IPacket packet)
        {
            RegisterRequest message = (RegisterRequest)packet;
            var messageWrite = new ServerMessageWrite(message.ProtocolId(), message);
            var json = JsonConvert.SerializeObject(messageWrite);
            buffer.WriteString(json);
        }

        public IPacket Read(ByteBuffer buffer)
        {
            var packet = RegisterRequest.ValueOf();

            return packet;
        }
    }
}

File: UnityMMOClient/Lua/application/app/net/webSocket/luaProtocol/register/registerRequest.lua
---@class RegisterRequest
local RegisterRequest = class("RegisterRequest")
function RegisterRequest:ctor()
    --- 账号
    ---@type string
    self.account = string.empty
    --- 确认密码
    ---@type string
    self.affirmPassword = string.empty
    --- 密码
    ---@type string
    self.password = string.empty
end

---@param account string 账号
---@param affirmPassword string 确认密码
---@param password string 密码
---@return RegisterRequest
function RegisterRequest:new(account, affirmPassword, password)
    self.account = account --- java.lang.String
    self.affirmPassword = affirmPassword --- java.lang.String
    self.password = password --- java.lang.String
    return self
end

---@return number
function RegisterRequest:protocolId()
    return 1005
end

---@return string
function RegisterRequest:write()
    local message = {
        protocolId = self:protocolId(),
        packet = {
            account = self.account,
            affirmPassword = self.affirmPassword,
            password = self.password
        }
    }
    local jsonStr = JSON.encode(message)
    return jsonStr
end

---@return RegisterRequest
function RegisterRequest:read(data)

    local packet = self:new(
            data.account,
            data.affirmPassword,
            data.password)
    return packet
end

--- 账号
---@type  string 账号
function RegisterRequest:getAccount()
    return self.account
end

--- 确认密码
---@type  string 确认密码
function RegisterRequest:getAffirmPassword()
    return self.affirmPassword
end

--- 密码
---@type  string 密码
function RegisterRequest:getPassword()
    return self.password
end



return RegisterRequest
